### PR TITLE
Resolves #13: Shows example usage of IceNet library with Pytorch UNet and cGAN models

### LINKED ISSUE #13: Show usage of pytorch using IceNet library
Relates to following issues:
* icenet-ai/icenet-project#47
* icenet-ai/icenet-project#54
* icenet-ai/icenet#211

Show examples using icenet's dataloader, and output pipeline in a jupyter notebook.

### PR DESCRIPTION
**Pull request by bnubald**
Resolves #13 

Adds two notebooks showing the usage of the IceNet library for @ampersandmcd 's [GAN](https://github.com/ampersandmcd/icenet-gan).

Notebooks (Both based on Andrew's original [code](https://github.com/ampersandmcd/icenet-gan) which in turn was based on Tom Andersson's original paper's [code](https://github.com/tom-andersson/icenet-paper)):
1. 1_icenet_forecast_unet.ipynb: PyTorch UNet implementation, similar to current Tensorflow implementation in IceNet.
2. 2_icenet_forecast_cgan.ipynb: PyTorch cGAN implementation, using UNet as generator.

The original code has been refactored to match current state of IceNet:
* Using IceNet for data download and processing
* Showing usage of new `IceNetDataSetPyTorch` class for interfacing existing code with PyTorch:
 ```from icenet.data.dataset import IceNetDataSetPyTorch```
* Daily predictions instead of monthly averaged
* Binary sea ice concentration (SIC) instead of original triple class (no ice, marginal, and full)
* Update and add metrics for binary classification
* Using IceNet masks
* Using IceNet for netCDF output of predictions and plotting

It also includes an environment file (`pytorch/environment.yml`) which will help set up both Tensorflow and PyTorch together with CUDA support in a conda environment on a Linux system (tested on BAS HPC). This should make it easier to run through these notebooks as all dependencies should be covered (including IceNet library [v0.2.8_dev](https://github.com/icenet-ai/icenet/tree/v0.2.8_dev)).

Thanks to @ampersandmcd and @rchan26 for covering most of the leg work!

### CODE REVIEW

**Review by JimCircadian — APPROVED** [review]
Excellent, thanks for structuring the files etc nicely it definitely makes things nicer to look at!

**Comment by bnubald** [comment]
Relates to icenet-ai/icenet#225
